dhcpv6/dhcpv6message.go

```diff
@@ -161,17 +161,23 @@ func NewRequestFromAdvertise(adv *Message, modifiers ...Modifier) (*Message, err
 }
 
 // NewReplyFromMessage creates a new REPLY packet based on a
-// Message. The function is to be used when generating a reply to
-// REQUEST, CONFIRM, RENEW, REBIND, RELEASE and INFORMATION-REQUEST packets.
+// Message. The function is to be used when generating a reply to a SOLICIT with
+// rapid-commit, REQUEST, CONFIRM, RENEW, REBIND, RELEASE and INFORMATION-REQUEST
+// packets.
```

Comment:
I wonder if we should merge `NewReplyFromMessage` and `NewAdvertiseFromSolicit` to avoid having to figure out which one to use.

Reply:
I think not - NewReplyFromMessage explicitly builds a Reply from any message, and fails if it's not acceptable by the protocol. NewAdvertiseFromSolicit instead is specific about solicit, and only to be used for the "old" four-way transactions. Collapsing Advertise/Reply from Solicit to me sounds more complicated

Reply:
I am thinking about a server handler when it receives a solicit: How should it build a response? Using `NewReplyFromMessage` or `NewAdvertiseFromSolicit`? Should the server handler check if the solicit has a rapid commit option? If so it will duplicate what we are already doing here. Why not merging them instead and having a function that takes a solicit and returns whatever needs to be returned.

Reply:
I made this PR exactly for that reason. The server handler does something like
```
if msg.GetOneOption(OptionRapidCommit) {
    resp, err = dhcpv6.NewReplyFromMessage(msg)
} else {
  resp, err = dhcpv6.NewAdvertiseFromSolicit(msg)
}
```
At some point in the future I see NewAdvertiseFromSolicit going away, but until then this seems the straightforward approach